luminoth/tools/checkpoint/__init__.py

```diff
@@ -0,0 +1,725 @@
+
+
+@click.command(help='Remove a checkpoint from the index and delete its files.')
+@click.argument('id_or_alias')
+def delete(id_or_alias):
+    db = read_checkpoint_db()
+    checkpoint = get_checkpoint(db, id_or_alias)
+    if not checkpoint:
+        click.echo(
+            "Checkpoint '{}' not found in index.".format(id_or_alias)
+        )
+        return
+
+    # If checkpoint is local, remove entry from index. If it's remote, only
+    # mark as ``NOT_DOWNLOADED``.
+    if checkpoint['source'] == 'local':
+        db['checkpoints'] = [
+            cp for cp in db['checkpoints']
+            if not cp['id'] == checkpoint['id']
+        ]
+    else:  # Remote.
+        if checkpoint['status'] == 'NOT_DOWNLOADED':
+            click.echo("Checkpoint isn't downloaded. Nothing to delete.")
+            return
+        checkpoint['status'] = 'NOT_DOWNLOADED'
+    save_checkpoint_db(db)
+
+    # Delete tar file associated to checkpoint.
+    path = get_checkpoint_path(checkpoint['id'])
+    try:
+        shutil.rmtree(path)
+    except OSError:
+        # The tar is not present, warn the user just in case.
+        click.echo(
+            'Skipping files deletion; not present in {}.'.format(path)
+        )
+
+    click.echo('Checkpoint {} deleted successfully.'.format(checkpoint['id']))
+
+
+@click.command(help='Export a checkpoint to a tar file for easy sharing.')
+@click.argument('id_or_alias')
+@click.option('--output', default='.', help="Specify the output location.")
+def export(id_or_alias, output):
+    db = read_checkpoint_db()
+    checkpoint = get_checkpoint(db, id_or_alias)
+    if not checkpoint:
+        click.echo(
+            "Checkpoint '{}' not found in index.".format(id_or_alias)
+        )
+        return
+
+    # Create the tar that will contain the checkpoint.
+    tar_path = os.path.join(
+        os.path.abspath(os.path.expanduser(output)),
+        '{}.tar'.format(checkpoint['id'])
+    )
+    checkpoint_path = get_checkpoint_path(checkpoint['id'])
+    with tarfile.open(tar_path, 'w') as f:
+        # Add the config file. Dump the dict into a BytesIO, go to the
+        # beginning of the file and pass it as a file to the tar.
+        metadata_file = six.BytesIO()
+        metadata_file.write(json.dumps(checkpoint).encode('utf-8'))
+        metadata_file.seek(0)
+
+        tarinfo = tarfile.TarInfo(name='metadata.json')
+        tarinfo.size = len(metadata_file.getvalue())
+        f.addfile(tarinfo=tarinfo, fileobj=metadata_file)
+
+        metadata_file.close()
+
+        # Add the files present in the checkpoint's directory.
+        for filename in os.listdir(checkpoint_path):
+            path = os.path.join(checkpoint_path, filename)
+            f.add(path, filename)
+
+    click.echo('Checkpoint {} exported successfully.'.format(checkpoint['id']))
+
+
+@click.command(help='Import a checkpoint tar into the local index.')
+@click.argument('path')
+def import_(path):
```

Comment:
Shouldn't this also accept folders? Otherwise it makes it compulsory to get your checkpoints as tars from running `export`, while some users may prefer sharing them in as folders. For example they may want to share their whole `~/.luminoth/checkpoints` folder with another person.

Maybe we should completely hide the usage of tars from the user and remove the `export` command? Not sure about this idea though.